server/container/spring/src/main/java/org/apache/james/container/spring/osgi/AbstractBundleTracker.java

```diff
@@ -88,7 +88,7 @@ public void bundleChanged(BundleEvent event) {
                     URL e = (URL) entrs.nextElement();
                     String file = e.getFile();
 
-                    String className = file.replaceAll("/", ".").replaceAll(".class", "").replaceFirst(".", "");
+                    String className = file.replaceAll("/", ".").replaceAll("\\.class", "").replaceFirst("\\.", "");
```

Comment:
no test?

Reply:
It's hard to test an abstract class

Reply:
btw I see it"s located in an osgi package, is it used at all?

Reply:
I have no idea, should I drop all the file?

Reply:
remove it and see if it still pass the CI?

Reply:
not used => dropping